### DISCUSSION TITLE
error NU1201: Project [CppProjChild] is not compatible with net6.0-windows10.0.19041 (.NETCoreApp,Version=v6.0). Project [CppProjChild] supports: native (native,Version=v0.0)

### DISCUSSION THREAD

**Discussion by whodges** [NuGet/Home#11889] [Q&A] [answered]
Hello,

We're using Visual Studio 2022 on a solution that we've recently converted from .NET Framework 4.6.1 to .NET 6.0.  It's a mix of C++ and C# projects, but with this particular build error, we have a C++ project (call it CppProjParent) that depends on another C++ project (call it CppProjChild) as a ProjectReference (both are in the same solution).  CppProjChild does NOT  have any CLR support, while CppProjParent DOES have .NET Core runtime support (net6.0-windows10.0).

I can build CppProjChild and CppProjParent on my local machine.  So can a coworker.  However, when we attempt to build these two projects on our TeamCity machines (using VS Build Tools 2022), CppProjParent fails with the following:

error NU1201: Project [CppProjChild] is not compatible with net6.0-windows10.0.19041 (.NETCoreApp,Version=v6.0). Project [CppProjChild] supports: native (native,Version=v0.0)

What we suspect is happening is that MSBuild is getting confused and trying to resolve nuget packages when building CppProjChild, but it's not configured for nuget, and, as such, is generating the error.  But we could be wrong.  Wanted to see if we're missing something as this appears to be an environmental problem.

As an experiment based on searches online, we tried setting ReferenceOutputAssembly to false in the ProjectReference to CppProjChild in CppProjParent.  While that gets rid of the NU1201 error, it seems to simply punt the issue down the road, and the build then fails on CppChildParent with:

 C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\MSBuild\Microsoft\NuGet\17.0\Microsoft.NuGet.targets(198,5): error : Sequence contains no elements [F:\TeamcityBuildAgent\work\37d9f5763d9c1e0d\Source\CppChildProj\CppChildProj.vcxproj]

Any thoughts/comments would be greatly appreciated.  Thanks!

**Comment by nkolev92** [comment, marked as answer]
@whodges 

I think you're looking for the feature in https://github.com/NuGet/Home/issues/8195. 

It's something that's in progress and should be out in 17.3. 
Are you using the preview versions of VS?

cc @lukka

Resolved, see below thread: 

tldr; The head project is  accidentally getting opted into PackageReference. 

The PackageReference opt ins are a PackageReference item or setting RestoreProjectStyle to PackageReference.

**Reply by doxxx** [reply]
I'm the coworker mentioned above. We're using normal released version of VS2022. Full IDE on our desktops, Build Tools on the Teamcity build agent.

I'm not sure that it's 8195. MSBuild/NuGet seems to be assuming that CppProjChild is a .NET/CLR project. But it's just a normal non-CLR C++ project. NuGet shouldn't even be looking at it, as far as I'm concerned.

**Reply by nkolev92** [reply]
> seems to be assuming that CppProjChild is a .NET/CLR project. But it's just a normal non-CLR C++ project. NuGet shouldn't even be looking at it, as far as I'm concerned.

The error message suggests NuGet looks at that project as a native one, hence:
> Project [CppProjChild] supports: native (native,Version=v0.0)

**Reply by doxxx** [reply]
Is there way to make NuGet ignore it?

**Reply by nkolev92** [reply]
ReferenceOutputAssembly ignores the project completely. 

PackageReference in CPP/CLI projects is something that's actively being worked on.

**Reply by doxxx** [reply]
When we try to use `ReferenceOutputAssembly` in the non-CLR C++ projects, we get this error:

```
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\MSBuild\Microsoft\NuGet\17.0\Microsoft.NuGet.targets(198,5): error : Sequence contains no elements [F:\TeamcityBuildAgent\work\37d9f5763d9c1e0d\Source\CppChildProj\CppChildProj.vcxproj]
```

We're not currently using `PackageReference` in the CLR C++ project (`CppProjParent` in our examples). We use `packages.config` to reference some NuGet assemblies, and a `ProjectReference` to `CppChildProj`.

**Reply by nkolev92** [reply]
The error that you're seeing is only going to be raised if NuGet thinks the project in question is PackageReference. 

The coded warnings only appear in PackageReference. 

The opt ins are `RestoreProjectStyle` and a PackageReference item existing. 
You can use the msbuild binary log (and the viewer) to see how/why your project is getting opted in.

I'd be happy to take a look as well, but you'd have to upload the binlog here.

**Reply by doxxx** [reply]
There are other C# projects in the solution which ProjectReference the projects we're discussing here, and those C# projects use PackageReference to pull in other unrelated dependencies. Would that trigger it?

**Reply by doxxx** [reply]
Thanks for the tip about the msbuild binlog. I'll take a look.

**Reply by nkolev92** [reply]
Each project needs to opt in individually, so it'd be in that exact project.

**Reply by doxxx** [reply]
We've got a Directory.Build.props, which adds a PackageReference to Microsoft.SourceLink.Bitbucket.Git. That's probably where it's coming from.

**Reply by doxxx** [reply]
Making the SourceLink PackageReference in Directory.Build.props conditional so that it only applies to .csproj seems to have resolved this particular issue. Thanks!
